ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/V1/Args.hs

```diff
@@ -0,0 +1,100 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Ouroboros.Consensus.Storage.LedgerDB.V1.Args (
+    BackingStoreArgs (..)
+  , FlushFrequency (..)
+  , LedgerDbFlavorArgs (..)
+  , QueryBatchSize (..)
+  , defaultLedgerDbFlavorArgs
+  , defaultQueryBatchSize
+  , defaultShouldFlush
+  ) where
+
+import           Control.Monad.IO.Class
+import qualified Data.SOP.Dict as Dict
+import           Data.Word
+import           GHC.Generics
+import           NoThunks.Class
+import           Ouroboros.Consensus.Storage.LedgerDB.V1.BackingStore.API
+import           Ouroboros.Consensus.Storage.LedgerDB.V1.BackingStore.Impl.LMDB
+import           Ouroboros.Consensus.Util.Args
+
+{-------------------------------------------------------------------------------
+  Arguments
+-------------------------------------------------------------------------------}
+
+-- | The /maximum/ number of keys to read in a backing store range query.
+--
+-- When performing a ledger state query that involves on-disk parts of the
+-- ledger state, we might have to read ranges of key-value pair data (e.g.,
+-- UTxO) from disk using backing store range queries. Instead of reading all
+-- data in one go, we read it in batches. 'QueryBatchSize' determines the size
+-- of these batches.
+--
+-- INVARIANT: Should be at least 1.
+--
+-- It is fine if the result of a range read contains less than this number of
+-- keys, but it should never return more.
+data QueryBatchSize =
+    -- | A default value, which is determined by a specific 'DiskPolicy'. See
+    -- 'defaultDiskPolicy' as an example.
+    DefaultQueryBatchSize
+    -- | A requested value: the number of keys to read from disk in each batch.
+  | RequestedQueryBatchSize Word64
+
+    -- | To disable queries, to be used in tests
+  | DisableQuerySize
+  deriving (Show, Eq, Generic)
+  deriving anyclass NoThunks
+
+defaultQueryBatchSize :: QueryBatchSize -> Word64
```

Comment:
`default*` seems like the wrong name for this, since `RequestedQueryBatchSize` is not a default.